pulp_python/app/tasks/sync.py

```diff
@@ -112,11 +114,22 @@ async def run(self):
         """
         # Bandersnatch includes leading slash when forming API urls
         url = self.remote.url.rstrip("/")
+        downloader = self.remote.get_downloader(url=url)
+        if not isinstance(downloader, HttpDownloader):
+            raise ValueError("Only HTTP(S) is supported for python syncing")
+
         async with Master(url) as master:
```

Comment:
It seems like Bandersnatch `Master()` has a proxy option you can pass in, but no auth.  I assume auth isn't terribly relevant in the ecosystem especially given the primary purpose is just mirroring PyPI.

Reply:
Yes, but it doesn't have proxy auth, so I think the partial patching is still the best way to go.